torch_xla/debug/metrics_test_wrapper.py

```diff
@@ -0,0 +1,84 @@
+"""Wrap PyTorch XLA jobs with extra logic around XLA metrics.
+
+It can be used to save a historical record of metrics and/or check metrics
+against a golden set of metrics to search for performance regressions.
+
+Example usage:
+python metrics_test_wrapper.py --metrics_output_filename="metrics.txt" \
+    -- python test/test_train_mnist.py --num_epochs=1
+"""
+import argparse
+import os
+import subprocess
+import sys
+
+import torch_xla.utils.gcsfs as gcsfs
+
+
+_CLOUD_STORAGE_PREFIX = 'gs://'
+_TMP_FILENAME = '/tmp/pytorch_tpu_metrics_tmp'
```

Comment:
Do not bolt file names like that. Generate with Python API.
Also, in MP replication mode, many files (one per core) will be created.
Create a temp folder (using Python tempfile API), and let the process write there.
Then remove on exit.

Reply:
Good call, done. Added a flag for distributed so we know where to read - LMK if there's a better way